system/Common.php

```diff
@@ -270,8 +270,11 @@ function esc($data, $context = 'html', $encoding = null)
 				$method = 'escape' . ucfirst($context);
 			}
 
-			// @todo Optimize this to only load a single instance during page request.
-			$escaper = new \Zend\Escaper\Escaper($encoding);
+                       static $escaper;
+                       if (! $escaper)
+                       {
+			        $escaper = new \Zend\Escaper\Escaper($encoding);
```

Comment:
how about pull `$encoding` (default uses 'utf-8') from `App`configuration ?

Reply:
I think that if `$encoding` is null than, yes, pulling it from `App` is appropriate.